segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/S3ClientWrapper.java

```diff
@@ -0,0 +1,264 @@
+import com.emc.object.s3.bean.GetObjectResult;
+import com.emc.object.s3.bean.ListObjectsResult;
+import com.emc.object.s3.bean.PutObjectResult;
+import com.emc.object.s3.bean.S3Object;
+import com.emc.object.s3.jersey.S3JerseyClient;
+import com.emc.object.s3.request.CompleteMultipartUploadRequest;
+import com.emc.object.s3.request.CopyPartRequest;
+import com.emc.object.s3.request.DeleteObjectsRequest;
+import com.emc.object.s3.request.PutObjectRequest;
+import com.emc.object.s3.request.SetObjectAclRequest;
+import io.pravega.common.io.StreamHelpers;
+import io.pravega.segmentstore.contracts.BadOffsetException;
+import io.pravega.segmentstore.storage.impl.extendeds3.AclSize;
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import lombok.SneakyThrows;
+import lombok.Synchronized;
+
+/**
+ * Client wrapper for S3JerseyClient.
+ */
+class S3ClientWrapper extends S3JerseyClient {
+    private final ConcurrentMap<String, AclSize> aclMap;
+    private final String baseDir;
+    private final ConcurrentMap<String, ConcurrentMap<Integer, CopyPartRequest>> multipartUploads = new ConcurrentHashMap<>();
+
+    public S3ClientWrapper(S3Config s3Config, ConcurrentMap<String, AclSize> aclMap, String baseDir) {
+        super(s3Config);
+        this.aclMap = aclMap;
+        this.baseDir = baseDir;
+    }
+
+    @Override
+    public DeleteObjectsResult deleteObjects(DeleteObjectsRequest request) {
+        return new DeleteObjectsResult();
+    }
+
+    @Synchronized
+    @Override
+    public PutObjectResult putObject(PutObjectRequest request) {
+
+        if (request.getObjectMetadata() != null) {
+            request.setObjectMetadata(null);
+        }
+        try {
+            Path path = Paths.get(this.baseDir, request.getBucketName(), request.getKey());
+            Files.createDirectories(path.getParent());
+            Files.createFile(path);
+        } catch (IOException e) {
+            throw new S3Exception(e.getMessage(), 0);
+        }
+        PutObjectResult retVal = new PutObjectResult();
+        if (request.getAcl() != null) {
+            long size = 0;
+            if (request.getRange() != null) {
+                size = request.getRange().getLast() + 1;
+            }
+            aclMap.putIfAbsent(request.getKey(), new AclSize(request.getAcl(),
+                    size));
+        }
+        return retVal;
+    }
+
+    @Override
+    @SneakyThrows(BadOffsetException.class)
+    public void putObject(String bucketName, String key, Range range, Object content) {
```

Comment:
Based on the other methods that are synchronized, I think this one is meant to be too.